datadog_test.go

```diff
@@ -0,0 +1,294 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package datadog
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"log"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	measureCount = stats.Int64("fooCount", "testing count metrics", stats.UnitBytes)
+	measureSum   = stats.Int64("fooSum", "testing sum metrics", stats.UnitBytes)
+	measureLast  = stats.Int64("fooLast", "testing LastValueData metrics", stats.UnitBytes)
+	measureDist  = stats.Int64("fooHisto", "testing histogram metrics", stats.UnitDimensionless)
+	testTags     []tag.Key
+)
+
+func newView(agg *view.Aggregation) *view.View {
+	return &view.View{
+		Name:        "fooCount",
+		Description: "fooDesc",
+		Measure:     measureCount,
+		Aggregation: agg,
+	}
+}
+
+func customNewView(measureName string, agg *view.Aggregation, tags []tag.Key, measure *stats.Int64Measure) *view.View {
```

Comment:
nit, but perhaps `newCustomView` is better since `new` is a common prefix for constructors.